src/main/kotlin/com/chocobar/reactiveKotlin/controller/QuestController.kt

```diff
@@ -23,10 +28,40 @@ class QuestController {
     @Autowired
     lateinit var questRepository: QuestRepository
 
+    val movieRequest = QuestFetcherImplementation()
+    val questFetcher=movieRequest.fetchQuest()
+    val questRemoteRepository = movieRequest.getRemoteRepository()
+
+    @GetMapping("/start")
+    suspend fun onStart() {
+        questFetcher.onCompletion {}
+            .catch {  }
+            .collect {
+
+            run {
+                    val generator = Generator(questRemoteRepository).guessOriginalLanguage()
+                generator.run {
+                    for(items in generator){
+                        val demoQuest= Quest(id = null,generator[items.id!!].text,generator[items.id!!].answer1,generator[items.id!!].answer2,generator[items.id!!].answer3,generator[items.id!!].answer4,generator[items.id!!].correctAnswer)
```

Comment:
podrias probar el forEach{} para bajar las lineas y armar un metodo que devuelva un Quest al que le pases el it(item) del forEach al generator.